Source/Diagnostics/Diagnostics.cpp

```diff
@@ -529,15 +529,24 @@ Diagnostics::InitBaseData ()
     m_all_field_functors.resize( nmax_lev );
 
     // For restart, move the m_lo and m_hi of the diag consistent with the
-    // current moving_window location
-    if (WarpX::do_moving_window) {
+    // current moving_window location. Account for start_moving_window_step and
+    // end_moving_window_step to only shift based on steps where the window was active.
+    if (WarpX::do_moving_window && warpx.getistep(0) > 0) {
```

Comment:
Why was `warpx.getistep(0) > 0` added here? 
Do we expect negative step number in WarpX?